rxtools/source/features/NandDumper.c

```diff
@@ -26,17 +26,17 @@ int NandSwitch(){
 	ConsoleShow();
 	while (true) {
         u32 pad_state = InputWait();
-		if(pad_state & BUTTON_X)    return  0;
-		if(pad_state & BUTTON_Y)    return  1;
-		if(pad_state & BUTTON_B) 	return -1;
+		if(pad_state & BUTTON_X) return SYS_NAND;
+		if(pad_state & BUTTON_Y) return EMU_NAND;
+		if(pad_state & BUTTON_B) return UNK_NAND;
     }
 }
 
 void NandDumper(){
 	File myFile;
-	int isEmuNand = checkEmuNAND() ? NandSwitch() : 0;
-	if(isEmuNand == -1) return;
-
+	int isEmuNand = SYS_NAND;
+	if(checkEmuNAND() && (isEmuNand = NandSwitch()) == UNK_NAND) return;
```

Comment:
NandSwitch() invoked only if checkEmuNAND() returned 1, i.e. emuNAND present. The save behaviour as before
No emunand - proceed with sysnand
Emunand present - ask which nand to use, is dialog cancelled - abort.